### PR TITLE
Added installation and running section.

### PR DESCRIPTION
**Pull request by Spring-dot**
## ReadMe: Add Installation and Running Section

Added installation and running section in the ReadMe.

Closes: #72 

### Describe the changes you've made

Added all steps required for installing and running this project in the local setup.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [ ] My code follows the guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly whereever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -125,6 +125,12 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
 > The typical MERN Stack involves *MongoDB* + ExpressJS + ReactJS + NodeJS. But, if you consider our product, it really needs a hierarchical database management system and not a document oriented one. MongoDB *doesn't perform well* under a lot of situations, so we're planning to use *MySQL (or MariaDB) or PostgreSQL* (We have already mentioned that it's yet to be decided between these two). I hope this note helps people understand why we are *not* using MongoDB in MERN Stack.
 
 ## Installation & Running
```
Comment (praveenscience) [inline ReadMe.md:127]: Need a line space.

File: ReadMe.md [hunk 2]
```diff
@@ -125,6 +125,12 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+1. Fork [this](https://github.com/praveenscience/Internship-LMS-FrontEnd) repository and Clone it to your local system.
```
Comment (praveenscience) [inline ReadMe.md:128]: No need of forking for people who want to just run, not contribute. Just cloning is enough.

File: ReadMe.md [hunk 3]
```diff
@@ -125,6 +125,12 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
 > The typical MERN Stack involves *MongoDB* + ExpressJS + ReactJS + NodeJS. But, if you consider our product, it really needs a hierarchical database management system and not a document oriented one. MongoDB *doesn't perform well* under a lot of situations, so we're planning to use *MySQL (or MariaDB) or PostgreSQL* (We have already mentioned that it's yet to be decided between these two). I hope this note helps people understand why we are *not* using MongoDB in MERN Stack.
 
 ## Installation & Running
+1. Fork [this](https://github.com/praveenscience/Internship-LMS-FrontEnd) repository and Clone it to your local system. 
+2. Check if Node is installed in your Operating system.
+ * To check if Node is installed, open the command prompt or other command line tool and type 'node -v'. If it prints a version number, it means Node is successfully installed.
+ * If the above command gives error, it means Node is not installed. You can download Node.js from https://nodejs.org/en/ and repeat previous step after installation.
```
Comment (praveenscience) [inline ReadMe.md:131]: Convert it into a link.

File: ReadMe.md [hunk 4]
```diff
@@ -125,6 +125,12 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+3. Navigate to the project directory and run 'npm install' to install all the dependencies.
```
Comment (praveenscience) [inline ReadMe.md:132]: Be clear here. Let's use `client` and also use back ticks for any code.

File: ReadMe.md [hunk 5]
```diff
@@ -125,6 +125,12 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+4. Run 'npm start' to start the application.
```
Comment (praveenscience) [inline ReadMe.md:133]: Backticks for code.

File: ReadMe.md (hunk 3 above)
Comment (praveenscience) [inline ReadMe.md:131]: Can we use a separate section called trouble shooting, a one level down heading and tell this? Oh wait, no, where's the system requirements part? 🤔 That should come under the installation right?

You totally ignored this: 😔🤦🏻‍♂️

image

**Review by praveenscience — CHANGES_REQUESTED** [review]
Good first attempt. Please resolve the review changes.

File: ReadMe.md (hunk 5 above)
Comment (kirtirajsinh) [reply ReadMe.md:133]: you should also mention that after npm install. navigate to the client folder `cd client` and then run `npm start`.

Comment (praveenscience) [reply ReadMe.md:133]: Thanks for that @kirtirajsinh. 😊

**Comment by praveenscience** [comment]
@Spring-dot It's been more than 3 days. Any update on this? Or should we give it to another person? Please let me know if you need any help on this?

**Comment by Spring-dot** [comment]
@praveenscience  I am facing some issues, I will let you know after 4 pm IST, will try to update it asap then.

**Comment by Spring-dot** [comment]
@praveenscience I have done few changes, please review and let me know what else to add or change.

File: ReadMe.md [hunk 6]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+ * NodeJS: This allows to run JavaScript on a computer server.
+
+ * Visual Studio Code: Text-editor for our project as we are using React. Add these extensions for ReactJS development - 
+   * ES7 React/Redux/GraphQL/React-Native snippets 
+   * GitLens - Git supercharged 
+   * Git History
+   * ESLint 
+   * Prettier - Code formatter
+
+ * PostMan: This software used as a REST Client to create and execute queries.
```
Comment (praveenscience) [inline ReadMe.md:157]: Remove the Line Spaces here. This looks really nice. Make the text a bit clearer.

File: ReadMe.md [hunk 7]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
-1. Fork [this](https://github.com/praveenscience/Internship-LMS-FrontEnd) repository and Clone it to your local system. 
-2. Check if Node is installed in your Operating system.
- * To check if Node is installed, open the command prompt or other command line tool and type 'node -v'. If it prints a version number, it means Node is successfully installed.
- * If the above command gives error, it means Node is not installed. You can download Node.js from https://nodejs.org/en/ and repeat previous step after installation.
-3. Navigate to the project directory and run 'npm install' to install all the dependencies.
-4. Run 'npm start' to start the application.
+
+**System Requirements**
+
+For Server/dev:
```
Comment (praveenscience) [inline ReadMe.md:142]: Try?
```suggestion
For Server Side Setup / Development Environment:
```

File: ReadMe.md [hunk 8]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+
+ * Visual Studio Code: Text-editor for our project as we are using React. Add these extensions for ReactJS development - 
+   * ES7 React/Redux/GraphQL/React-Native snippets 
+   * GitLens - Git supercharged 
+   * Git History
+   * ESLint 
+   * Prettier - Code formatter
+
+ * PostMan: This software used as a REST Client to create and execute queries. 
+
```
Comment (praveenscience) [inline ReadMe.md:158]: Let's add a client side requirements. Just a decent modern browser supporting latest technologies is enough. Check React's client requirements, browser requirements.

File: ReadMe.md [hunk 9]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+**Installation And Setup**
```
Comment (praveenscience) [inline ReadMe.md:159]: Capitalisation.
```suggestion
**Installation and Setup**
```

File: ReadMe.md [hunk 10]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+   cd Internship-LMS-Frontend
+```
+3. Navigate to the client folder
+```
+   cd client
+```
+4. Install all the dependencies
+```
+   npm install
+```
```
Comment (praveenscience) [inline ReadMe.md:176]: No need of spaces in front of the code. Also use syntax highlighting like: <code>```bash</code>.

File: ReadMe.md [hunk 11]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+
+  **Trouble Shooting**
+
+   * If you get error while installing all the dependencies, check if Node is installed using command line tool
+   ```
+      node -v
```
Comment (praveenscience) [inline ReadMe.md:182]: Can we also add `npm` here?

File: ReadMe.md [hunk 12]
```diff
@@ -136,12 +136,68 @@ We are primarily considering JavaScript (sorry, not TypeScript) as the base lang
...
+   cd Internship-LMS-Frontend
+```
+2. Navigate to the client folder
+```
+   cd client
+```
+3. Run the application
+```
+   npm start
+```
```
Comment (praveenscience) [inline ReadMe.md:200]: No need of spaces in front of the code. Also use syntax highlighting like: <code>```bash</code>.

**Review by praveenscience — CHANGES_REQUESTED** [review]
Very good attempt. Couple of changes, please address them. Thanks a lot.

File: ReadMe.md (hunk 8 above)
Comment (Spring-dot) [reply ReadMe.md:158]: ok

File: ReadMe.md (hunk 12 above)
Comment (niya210) [reply ReadMe.md:200]: Are you Done with it @Spring dot? Or I should do it??

Comment (Spring-dot) [reply ReadMe.md:200]: > Are you Done with it @spring dot? Or I should do it??

Yes, will commit changes now

**Review by praveenscience — DISMISSED** [review]
👍

**Review by praveenscience — APPROVED** [review]
👍

**Pull request merged**